test/unit/org/apache/cassandra/cql3/functions/FormatTimeFctTest.java

```diff
@@ -0,0 +1,327 @@
+
+        assertRows(execute("select format_time(col1), format_time(col2), format_time(col3) from %s where pk = 1"),
+                   row(null, null, null));
+
+        assertRows(execute("select format_time(col1, 's') from %s where pk = 1"), row((Object) null));
+        assertRows(execute("select format_time(col1, 's', 'd') from %s where pk = 1"), row((Object) null));
+    }
+
+    @Test
+    public void testHandlingNullArguments()
+    {
+        createTable(of(TEXT, ASCII, INT),
+                    new Object[][]{ { 1, null, null, null },
+                                    { 2, "'1'", "'2'", 3 } });
+
+        assertRows(execute("select format_time(col1, null) from %s where pk = 1"), row((Object) null));
+
+        for (String functionCall : new String[] {
+        "format_time(col3, null)",
+        "format_time(col3, null, null)",
+        "format_time(col3, null, 'd')",
+        "format_time(col3, 'd', null)"
+        })
+        {
+            assertThatThrownBy(() -> execute("select " + functionCall + " from %s where pk = 2"))
+            .isInstanceOf(InvalidRequestException.class)
+            .hasMessageContaining("none of the arguments may be null");
+        }
+    }
+
+    @Test
+    public void testFuzzRandomGenerators()
+    {
+        createTable("CREATE TABLE %s (pk int primary key, col1 int)");
+        qt().withExamples(1024).forAll(integers().allPositive()).checkAssert(
+        (randInt) -> {
+            execute("INSERT INTO %s (pk, col1) VALUES (?, ?)", 1, randInt);
+            assertRows(execute("select format_time(col1, 's', 'm') from %s where pk = 1"), row(format((double) randInt * (1 / 60.0)) + " m"));
+            assertRows(execute("select format_time(col1, 's', 'h') from %s where pk = 1"), row(format((double) randInt * (1 / 3600.0)) + " h"));
+            assertRows(execute("select format_time(col1, 's', 'd') from %s where pk = 1"), row(format((double) randInt * (1 / 86400.0)) + " d"));
+            assertRows(execute("select format_time(col1, 'ms', 'm') from %s where pk = 1"), row(format((double) randInt * (1 / (60 * 1000.0))) + " m"));
+            assertRows(execute("select format_time(col1, 'ms', 'h') from %s where pk = 1"), row(format((double) randInt * (1 / (3600 * 1000.0))) + " h"));
+            assertRows(execute("select format_time(col1, 'ms', 'd') from %s where pk = 1"), row(format((double) randInt * (1 / (86400 * 1000.0))) + " d"));
+        });
+    }
+
+    private void createTable(List<CQL3Type.Native> columnTypes, Object[][] rows)
+    {
+        String[][] columns = new String[columnTypes.size() + 1][2];
+
+        columns[0][0] = "pk";
+        columns[0][1] = "int";
+
+        for (int i = 1; i <= columnTypes.size(); i++)
+        {
+            columns[i][0] = "col" + i;
+            columns[i][1] = columnTypes.get(i - 1).name().toLowerCase();
+        }
+
+        createTable(columns, rows);
+    }
+
+    private void createDefaultTable(Object[][] rows)
```

Comment:
@chengw-netflix this code to the end of the file is same for the other test as well. Maybe having `AbstractFunctionTest` and moving it there as protected method would be better.